draft-bar-cfrg-spake2plus.md

```diff
@@ -193,60 +198,72 @@ attackers.
+It is essential that both Z and V be used in combination with the transcript to
 derive the keying material. The protocol transcript encoding is shown below.
 
 ~~~
-TT = len(A) || A || len(B) || B || len(X) || X
-  || len(Y) || Y || len(Z) || Z || len(V) || V
+TT = len(Context) || Context ||
+  || len(A) || A || len(B) || B
+  || len(M) || M || len(N) || N
+  || len(X) || X || len(Y) || Y
+  || len(Z) || Z || len(V) || V
   || len(w0) || w0
 ~~~
 
+Context is an application-specific customization string shared between both
+parties and SHOULD precede the remaining transcript. It might contain the
+name and version number of the higher-level protocol, or simply the name and version
+number of the application. The context MAY include additional data such as the
+chosen ciphersuite and PBKDF parameters like the iteration count or salt.
+The context and its length prefix MAY be omitted.
+
 If an identity is absent, it is omitted from the transcript entirely. For example,
-if both A and B are absent, then TT = len(X) || X || len(Y) || Y || len(Z) || Z || len(w0) || w0.
-Likewise, if only A is absent, TT = len(B) || B || len(X) || X || len(Y) || Y || len(Z) || Z || len(w0) || w0.
+if both A and B are absent, then TT = len(Context) || Context || len(M) || M || len(N) || N || len(X) || X || len(Y) || Y || len(Z) || Z || len(w0) || w0.
+Likewise, if only A is absent, TT = len(Context) || Context || len(B) || B || len(M) || M || len(N) || N || len(X) || X || len(Y) || Y || len(Z) || Z || len(w0) || w0.
 This must only be done for applications in which identities are implicit. Otherwise,
 the protocol risks Unknown Key Share attacks (discussion of Unknown Key Share attacks
 in a specific protocol is given in {{?I-D.ietf-mmusic-sdp-uks}}.
 
-Upon completion of this protocol, A and B compute shared secrets Ke, KcA, and KcB as
-specified in {{keys}}. B MUST send A a key confirmation message Fb
-so both parties agree upon these shared secrets. This confirmation message Fb
+Upon completion of this protocol, A and B compute shared secrets Ka, Ke, KcA,
+and KcB as specified in {{keys}}. B MUST send A a key confirmation message cB
+so both parties agree upon these shared secrets. This confirmation message cB
 is computed as a MAC over the received share (pA) using KcB. Specifically, B
-computes Fb = MAC(KcB, pA). After receipt and verification of B's confirmation
-message, A MUST send B a confirmation message using a MAC computed equivalently
-except with the use of pB and KcA. Key confirmation verification requires computing
-F and checking for equality against that which was received.
+computes cB = MAC(KcB, pA), where MAC is also a secure PRF. After receipt and
+verification of B's confirmation message, A MUST send B a confirmation message
+using a MAC computed equivalently except with the use of pB and KcA. Key
+confirmation verification requires recomputation of the MAC and checking
+for equality against that which was received.
```

Comment:
How do we deal with tail loss? What if an adversary drops the confirmation message? It's probably better to say, B MUST NOT send application data to A until it has received and verified the confirmation message.

Reply:
Let's add that, good point.